src/utils/cache.rs

```diff
@@ -62,20 +62,59 @@ pub trait CacheExt: Cache {
         client: &reqwest::blocking::Client,
         url: &String,
         timeout: usize,
-    ) -> Result<String> {
-        let data: String =
-            match self.fetch(&url) {
-                Some(cached) => cached,
-                None => {
-                    let response = client.get(url).send()?;
-                    // Don't cache if it is a Server error
-                    if response.status().is_server_error() { anyhow::bail!("Got server error for {}", url); };
-                    let raw_data = response.text()?;
-                    self.create(&url, &raw_data, timeout);
-                    raw_data
+    ) -> ApiResult<String> {
+        match self.fetch(&url) {
+            Some(cached) => {
+                let cached_with_code = CachedWithCode::split(&cached);
+                return if !cached_with_code.is_error() {
```

Comment:
Theoretically we could extract this into a method (building the response based on a boolean from a code + data)

Reply:
I added a method `to_result` in the `impl CacheWithCode` block.